week2-7.2-shipping_turtle.py

```diff
@@ -0,0 +1,139 @@
+            After creating a PO box and sending a letter,
+            the recipient should have 1 message in the
+            inbox.
+
+            po_box = PostOffice(['a', 'b'])
+            message_id = po_box.send_message('a', 'b', 'Hello!')
+            len(po_box.boxes['b'])
+            1
+            message_id
+            1
+        """
+        user_box = self.boxes[recipient]
+        self.message_id = self.message_id + 1
+        message_details = {
+            'id': self.message_id,
+            'body': message_body,
+            'sender': sender,
+        }
+        if urgent:
+            user_box.insert(0, message_details)
+        else:
+            user_box.append(message_details)
+        return self.message_id
+
+
+    """
+    read the messages of a certain user
+
+        Args:
+            user (str): the user's name.
+            N (int): number of messages wanted.
+        Returns:
+            list: N messages that the user has
+
+
+        Examples:
+            a user has number of messages and he wants 
+            to read them.
+            
+            po_box = PostOffice(['a', 'b'])
+            po_box.send_message('a', 'b', 'Hello!')
+            po_box.send_message('a', 'b', 'How are you?')
+            po_box.send_message('a', 'b', 'nahh bru')
+            po_box.send_message('a', 'b', 'you here?')
+            print(po_box.read_inbox('b'))
+            
+            ['Hello!', 'How are you?', 'nahh bru', 'you here?']
+            
+            print(po_box.read_inbox('b', 2))
+            
+            ['Hello!', 'How are you?']
+            
+            
+            
+            
+    """
+    def read_inbox(self, user, N = 0):
+        if N == 0:
+            N = len(self.boxes[user])
+
+        messages = []
+        result = []
+        for usr in self.boxes:
+            if usr == user:
+                messages = self.boxes[usr]
+        for msg in islice(messages, N):
+            result.append(msg['body'])
+
+        return result
+
+    """
+    read the inbox of a user depending on a given string
+
+        Args:
+            user (str): the user's name.
+            wanted (str): the string to search for in messages.
+        Returns:
+            list: all messages that contain wanted in them
+
+
+        Examples:
+            a user has number of messages and he wants 
+            look for messages containing 'you' in them
+
+            po_box = PostOffice(['a', 'b'])
+            po_box.send_message('a', 'b', 'Hello!')
+            po_box.send_message('a', 'b', 'How are you?')
+            po_box.send_message('a', 'b', 'nahh bru')
+            po_box.send_message('a', 'b', 'you here?')
+            print(po_box.search_inbox('b', 'you'))
+
+            ['How are you?', 'you here?']
+
+
+    """
+    def search_inbox(self, user, wanted):
+        messages = self.read_inbox(user)
+        result = []
+        for msg in messages:
+            if msg.lower().__contains__(wanted.lower()):
```

Comment:
what is msg? you need to check whether the body or title of the message have the wanted string.